xformers/components/attention/core.py

```diff
@@ -216,13 +220,44 @@ def scaled_dot_product_attention(
     v: torch.Tensor,
     att_mask: Optional[Union[AttentionMask, "SparseCS", torch.Tensor]],
     dropout: Optional[torch.nn.Module] = None,
+    num_heads: int = 1,
 ) -> torch.Tensor:
     autocast_disabled = (
         _is_sparse_available
         and isinstance(att_mask, SparseCS)
         or (att_mask is not None and att_mask.is_sparse)
     )
 
+    # Check if causal is required but mask is not sparse; if fp16 or under amp context
+    switch_to_blocksparse = (
+        _is_triton_available
+        and (att_mask is not None and not att_mask.is_sparse)  # redundant?
```

Comment:
not redundant, it's good I think. It's possible not to have a att_mask here, and it's also possible that there's a att_mask and it's sparse, in both cases you would not want to switch to blocksparse